src/org/traccar/protocol/OwnTracksProtocolDecoder.java

```diff
@@ -80,16 +85,29 @@ protected Object decode(
         if (root.containsKey("t")) {
             position.set("t", root.getString("t"));
         }
+        if (root.containsKey("p")) {
+            position.set("pb", root.getInt("p"));
+        }
         if (root.containsKey("batt")) {
             position.set(Position.KEY_BATTERY, root.getInt("batt"));
         }
-        if (root.containsKey("topic")) {
-            position.set("topic", root.getString("topic"));
-        }
 
         position.setTime(new Date(root.getJsonNumber("tst").longValue() * 1000));
 
-        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, root.getString("tid"));
+        Boolean haveTopic = root.containsKey("topic");
```

Comment:
You shouldn't use boxed primitives. Also, there is no need for variables here at all.